### DISCUSSION TITLE
Release 1.2 is out!

### DISCUSSION THREAD

**Discussion by Danielku15** [CoderLine/alphaTab#530] [General]
I'm happy to announce that I completed today all planned work items for version 1.2 and created a new 1.2 stable release. 

https://github.com/CoderLine/alphaTab/releases/tag/v1.2.0
https://www.npmjs.com/package/@coderline/alphatab
https://www.nuget.org/packages/AlphaTab/1.2.0
https://www.nuget.org/packages/AlphaTab.Windows/1.2.0

Feel free to report any issues you find in this release. 

# Release Notes
The next release of alphaTab is done. The theme of this release was: 

> Main focus will be improving the visual display by avoiding collisions and improving some visual elements.
 
We tried to avoid any breaking changes on the API level and upgrades should be possible without any code changes. 
But there are various behavior changes where you might want to adopt new settings, values etc.  

This release brings you:

## In alphaTab 1.2 we dropped support for Internet Explorer ❗
One of the biggest changes/decisions we took in version 1.2 is to drop support for Internet Explorer.
This allowed us to make various improvements in regards to performance and more modern code. 
We made [a poll](https://github.com/CoderLine/alphaTab/discussions/460) whether dropping IE support is generally accepted, but in case you missed it and have need for Internet Explorer support, reach out to us via [Discussions](https://github.com/CoderLine/alphaTab/discussions) and we will see what we can do. 

## New Features 💡 

- [General] JSON Serialization for the .net platform (Issue: #461, PR: #466, [Docs](https://www.alphatab.net/docs/guides/lowlevel-apis/#serialize-data-model-fromto-json)) 
<br/>

- [File Formats] Guitar Pro 7 Exporter (Issue: #443, PR: #447, [Docs](https://www.alphatab.net/docs/guides/exporter))
- [File Formats] Detect Lyrics from beat text for Guitar Pro 3-5 files (Issue: #402, PR: #506, [Docs](https://www.alphatab.net/docs/reference/settings/importer/beattextaslyrics/))
<br/>

## Improvements 🚀 

- [General] Respect original settings when printing and allow additional print settings (Issue: #469, PR: #486, [Docs](https://www.alphatab.net/docs/reference/api/print/))
- [General] Compilation of alphaTab to ES6 code for smaller footprint and more modern code (Issue: #508, PR: #512)
- [General] Smaller copyright watermark (PR: #512)
- [General] Removed various polyfills, usage of oudated APIs, better SVG generation (Issue: #509, PR: #514)
<br/>

- [File Formats] Respect track volume in Guitar Pro 7 files (Issue: #446, PR: #445 thanks to @jordanske for the contribution)
- [File Formats] Respect track volume and balance in MusicXML files (Issue: #451, PR: #454 thanks to @jordanske for the contribution)
- [File Formats] Auto detection of instruments and tunings in alphaTex (Issue: #484, PR: #449)
- [File Formats] Add support for full bar rests in Capella and MusicXML (Issue: #495, PR: #503)
<br />

- [Rendering] New Note Beaming algorithm which avoids collisions on rests (Issue: #296, PR: #491)
- [Rendering] Render multiple tunings in multi-track rendering cases (Issue: #463, PR: #492)
- [Rendering] New Grace Positioning Logic (Issues: #227, #335, PR: #493)
- [Rendering] Render HighDPI images for HTML5 Canvas (Issue: #496, PR: #497)
- [Rendering] Displace Rests in multi-voice scenarios (Issue: #355, PR: #502)
- [Rendering] Fixed various alignment issues of notes, tuplets and beams (PR: #502)
- [Rendering] Use of IntersectionObserver for faster lazy loading (Issue: #510, PR: #515)
- [Rendering] Use of ResizeObserver for faster/more-reliable resize detection (Issue: #511, PR: #516)
<br/>

- [Audio] Removed max volume limitation (Issue: #453, PR: #487)
- [Audio] More data on the `midiLoaded` event for synchronization with external audio sources (Discussion: #520, PR: #522)
- [Audio] New event fors for obtaining midi events like metronome (Issue: #450, PR: #523, [Docs](https://www.alphatab.net/docs/guides/handling-midi-events))

## Bug Fixes 🕷️ 

- [Rendering] Correct display of accidentals on tied notes across bars (Issue: #472, PR: #485)
- [Rendering] Various zoom related issues (Issue: #452, PR: #483)
- [Rendering] Added new property for WinForms control page background (replaces Foreground) (Issue: #503) 
If you had custom colors before, be sure to set the new `NotationBackColor` property. 
<br/>

- [Audio] Fixed wrong bend generations on tied bends (Issue: #470, PR: #488)

## Maintenance

- [General] Automatic generation of JSON serialization code (Issue: #461, PR: #466)
<br/>

**Comment by goodgame365** [comment]
very good
